cpp/src/arrow/engine/substrait/serde.h

```diff
@@ -133,13 +153,34 @@ ARROW_ENGINE_EXPORT Result<std::vector<acero::Declaration>> DeserializePlans(
 /// \param[out] ext_set_out if non-null, the extension mapping used by the Substrait
 /// Plan is returned here.
 /// \param[in] conversion_options options to control how the conversion is to be done.
-/// \return a vector of ExecNode declarations, one for each toplevel relation in the
-/// Substrait Plan
+/// \return an ExecNode corresponding to the single toplevel relation in the Substrait
+/// Plan
 ARROW_ENGINE_EXPORT Result<std::shared_ptr<acero::ExecPlan>> DeserializePlan(
     const Buffer& buf, const std::shared_ptr<dataset::WriteNodeOptions>& write_options,
     const ExtensionIdRegistry* registry = NULLPTR, ExtensionSet* ext_set_out = NULLPTR,
     const ConversionOptions& conversion_options = {});
 
+/// \brief Deserializes a single-relation Substrait Plan message into an execution plan
+///
+/// The output of the single Substrait relation will be written to a filesystem.
+/// `write_options_factory` can be used to control write behavior.
+///
+/// \param[in,out] plan an ExecNode corresponding to the single toplevel relation in the
+/// Substrait Plan
+/// \param[in] buf a buffer containing the protobuf serialization of a Substrait Plan
+/// message
+/// \param[in] write_options write options of a node consuming the batches produced by
+/// each toplevel Substrait relation
+/// \param[in] registry an extension-id-registry to use, or null for the default one.
+/// \param[out] ext_set_out if non-null, the extension mapping used by the Substrait
+/// Plan is returned here.
+/// \param[in] conversion_options options to control how the conversion is to be done.
+ARROW_ENGINE_EXPORT Status DeserializePlan(
```

Comment:
Why this change?

Reply:
Same rationale as in [this post](https://github.com/apache/arrow/pull/35513#discussion_r1188823725).